eclair-core/src/main/scala/fr/acinq/eclair/wire/LightningMessageTypes.scala

```diff
@@ -229,69 +229,150 @@ case class PerHopPayload(shortChannelId: ShortChannelId,
                          amtToForward: Long,
                          outgoingCltvValue: Long)
 
-// @formatter:off
+
+/**
+  * Optional TLV-based query message that can be appended to QueryShortChannelIds
+  * @param encoding 0 means uncompressed, 1 means compressed with zlib
+  * @param array array of query flags, each flags specifies the info we want for a given channel
+  */
+case class EncodedQueryFlags(encoding: EncodingType, array: List[Long]) extends Tlv
```

Comment:
I feel like the TLV should be defined in a separate file.

Reply:
Also this is not right: you should define a `sealed trait` for each TLV namespace